tests/test_activity.py

```diff
@@ -196,23 +233,26 @@ def test_instances_creation(monkeypatch, generators):
 
     monkeypatch.setattr(activity.Activity, '__init__', lambda self: None)
 
-    current_activity = activity.Activity()
-    current_activity.generators = generators
+    local_activity = activity.Activity()
+    external_activity = activity.ExternalActivity(timeout=60)
 
-    if len(current_activity.generators):
-        instances = list(current_activity.instances(dict()))
-        assert len(instances) == pow(10, len(generators))
-        for instance in instances:
-            assert isinstance(instance.local_context.get('i'), int)
+    for current_activity in [local_activity, external_activity]:
```

Comment:
The behavior of both should be the same if generators are applied. And this shows it works as expected.